### ISSUE TITLE
I think the dim of the torch.nn.functional.softmax() function does't work properly

### ISSUE THREAD

**Issue by czhaneva** [pytorch/pytorch#49841]
## To Reproduce

`import torch`
`import torch.nn.functional as F`
`A = torch.randn(2,3)`
`s1 = F.softmax(A, 1)`
`s2 = F.softmax(A.T, 0).T`
`print(s2 - s1)`

There is a gap between some elements.
I have manually calculated the result, the result is same with s1.
I have no idea why s1 is not equal to s2.

**Comment by ngimel** [comment]
I have the following results for your script:
```In [9]: a=torch.randn(2,3)                                                                                                                                                                      

In [10]: s1 = F.softmax(a, 1)                                                                                                                                                                   

In [11]: s2 = F.softmax(a.T, 0).T                                                                                                                                                               

In [12]: s1                                                                                                                                                                                     
Out[12]: 
tensor([[0.2751, 0.6759, 0.0490],
        [0.5515, 0.2382, 0.2102]])

In [13]: s2                                                                                                                                                                                     
Out[13]: 
tensor([[0.2751, 0.6759, 0.0490],
        [0.5515, 0.2382, 0.2102]])

In [14]: print(s2-s1)                                                                                                                                                                           
tensor([[ 2.9802e-08,  5.9605e-08,  0.0000e+00],
        [-5.9605e-08, -1.4901e-08, -1.4901e-08]])

```
The differences are on the order of floating point accuracy, that's expected.
Closing, please reopen if you still feel there's a bug.
